### ISSUE TITLE
undefined route

### ISSUE THREAD

**Issue by ghost** [ga-wdi-exercises/project3#68]
i'm trying to build my user routes and get the following error:

```
Error: Route.get() requires callback functions but got a [object Undefined]
    at /Users/thom/wdi/projects/campaign-finance/node_modules/express/lib/router/route.js:171:15
    at Array.forEach (native)
    at Route.(anonymous function) [as get] (/Users/thom/wdi/projects/campaign-finance/node_modules/express/lib/router/route.js:167:15)
    at Function.app.(anonymous function) [as get] (/Users/thom/wdi/projects/campaign-finance/node_modules/express/lib/application.js:406:19)
    at Object.<anonymous> (/Users/thom/wdi/projects/campaign-finance/index.js:30:5)
    at Module._compile (module.js:434:26)
    at Object.Module._extensions..js (module.js:452:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:475:10)
    at startup (node.js:118:18)
    at node.js:952:3
```

i've tried working on the routes in the user and candidate controllers, and think frankly i've lost track of where i am and how to get back to a reset.

my repo is https://github.com/thomhouseholder/campaign-finance and i'm working in the master branch.

**Comment by amaseda** [comment]
Can you paste the code for your router and user controller?

**Comment by ghost** [comment]
Yes!

router

```
var express = require('express');
var passport = require('passport');
var router = express.Router();
var usersController = require('../controllers/usersController');
var candidatesController = require('../controllers/candidatesController');


router.route('/')
  .get(candidatesController.home);


module.exports = router;
```

usersController

```
var passport = require("passport")

// GET /signup
function getSignup(request, response) {
  response.render("signup.hbs", { message: request.flash('signupMessage') });
}

// POST /signup
function postSignup(request, response) {
  var signupStrategy = passport.authenticate('local-signup', {
    successRedirect : '/',
    failureRedirect : '/signup',
    failureFlash : true
  });
  return signupStrategy(request, response);
}

// GET /login
function getLogin(request, response) {
  response.render('login.hbs', { message: request.flash('loginMessage') });
}

// POST /login
function postLogin(request, response) {
  var loginProperty = passport.authenticate('local-login', {
    successRedirect : '/',
    failureRedirect : '/login',
    failureFlash : true
  });
  return loginProperty(request, response);
}

// GET /logout
function getLogout(request, response) {
  request.logout();
  response.redirect('/');
}

// Restricted page
function secret(request, response){
  response.render("secret.hbs");
}

module.exports = {
  getLogin: getLogin,
  postLogin: postLogin ,
  getSignup: getSignup,
  postSignup: postSignup,
  getLogout: getLogout,
  secret: secret
};
```

**Comment by RobertAKARobin** [comment]
Your Candidates Controller has only one route defined: `home`:

``` js
function home(req, res) {
  res.render('index');
}

module.exports = {
  home: home

};
```

Since you have `app.get('/', candidatesController.index);`, it looks like a `index` route should also be defined.

**Comment by ghost** [comment]
So adding that route 

```

// GET /
function home(req, res) {
  res.render('index');
}
// 
function index(req, res) {
  res.render('index');
}

module.exports = {
  home: home
  index: index
};
```

now throws unexpected identifier.  am i exporting index correctly to use it elsewhere?

**Comment by amaseda** [comment]
Doesn't look like you're connecting `routes.js` to functions in your user controller. So if you want something like signup to work, you will need to create a route in `routes.js` that looks something like this...

``` js
router.route('/signup')
  .get(usersController.getSignup);
```

So checklist here is...
1. Have I created a route in `routes.js` that maps a URL (e.g., `/signup`) to a controller function (e.g., `usersController.getSignup`)?
2. Have I created a function in my controller (e.g., `function getSignup(req, res)`) that corresponds to the one referenced in `routes.js` (e.g., `usersController.getSignup`)?
